src/renderers/dom/client/eventPlugins/SimpleEventPlugin.js

```diff
@@ -634,7 +656,7 @@ var SimpleEventPlugin = {
   },
 
   willDeleteListener: function(inst, registrationName) {
-    if (registrationName === 'onClick') {
+    if (registrationName === 'onClick' && isInteractive(inst._tag)) {
```

Comment:
I had it reversed (sorry). It now reads `if (registrationName === 'onClick'
&& !isInteractive(inst._tag)) {`, inverting the isInteractive check.

Basically, iOS safari only exhibits the event bubbling issue for
non-interactive elements (like divs). Extra event handlers can be
eliminated by only attaching this event listener on those non-interactive
elements. Since the check was already created for the mouse events filter,
I went ahead and added it here.

This trades an event listener attachment for function invocation (that'll
probably get inlined), when I ran the numbers it seemed much faster
https://gist.github.com/nhunzaker/020335ee8ec0f1d487f97c7ddcfbd4a2.

On Fri, Sep 2, 2016 at 3:33 PM Ben Alpert <notifications@github.com> wrote:

> In src/renderers/dom/client/eventPlugins/SimpleEventPlugin.js
> <https://github.com/facebook/react/pull/7642#discussion_r77400816>:
>
> > @@ -634,7 +656,7 @@ var SimpleEventPlugin = {
> >    },
> >
> >    willDeleteListener: function(inst, registrationName) {
> > -    if (registrationName === 'onClick') {
> > +    if (registrationName === 'onClick' && isInteractive(inst._tag)) {
>
> I don't understand what this change is for. Can you explain?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebook/react/pull/7642/files/216f0c7461154336083649aae5dd6481295e758a#r77400816>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAkEOB61iktWlfL6kqfoypNnapaRNRs3ks5qmHnwgaJpZM4JzxX4>
> .
>